core/arch/x86_64/kernel/thread_optee_smc.c

```diff
@@ -720,17 +720,37 @@ void __noreturn sm_sched_nonsecure(void)
 #elif defined CFG_IVSHMEM
 #define OPTEE_HANDLE_DONE 0xa5a5a5a5
 
-extern struct thread_smc_args *g_smc_args;
-extern struct optee_smc_ring *smc_avail_ring;
-extern struct optee_smc_ring *smc_used_ring;
-extern struct optee_vm_ids *smc_vm_ids;
+extern struct thread_smc_args *g_smc_args[];
+extern struct optee_smc_ring *smc_avail_ring[];
+extern struct optee_smc_ring *smc_used_ring[];
+extern struct optee_vm_ids *smc_vm_ids[];
+extern uint8_t g_ivshmem_dev_num;
 
 static unsigned int smc_lock = SPINLOCK_UNLOCK;
+//TODO: need to consider SMP case
+uint8_t g_smc_idx = 0;
```

Comment:
this g_smc_idx is only used in sm_sched_nonsecure()? in this case, a local static variable will be better.
also consider hiding this variable to check_ree_requests() to make code clean

Reply:
consider make g_smc_idx as a static global variable which limits to this .c file, and expose get_cur_smc_idx() api for other .c files.